src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -0,0 +1,266 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Microsoft.Framework.Cache.Memory;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers
+{
+    /// <summary>
+    /// <see cref="TagHelper"/> implementation targeting &lt;cache&gt; elements.
+    /// </summary>
+    public class CacheTagHelper : TagHelper
+    {
+        private const string VaryByAttributeName = "asp-vary-by";
+        private const string VaryByHeaderAttributeName = "asp-vary-by-header";
+        private const string VaryByQueryAttributeName = "asp-vary-by-query";
+        private const string VaryByRouteAttributeName = "asp-vary-by-route";
+        private const string VaryByCookieAttributeName = "asp-vary-by-route";
+        private const string VaryByUserAttributeName = "asp-vary-by-user";
+        private const string ExpiresOnAttributeName = "asp-expires-on";
+        private const string ExpiresAfterAttributeName = "asp-expires-after";
+        private const string ExpiresSlidingAttributeName = "asp-expires-sliding";
+        private const string CachePriorityAttributeName = "asp-priority";
+        private const string CacheKeyTokenSeparator = "||";
+        private static readonly char[] AttributeSeparator = new[] { ',' };
+
+        [Activate]
+        public IMemoryCache MemoryCache { get; set; }
```

Comment:
Why public and not protected/protected internal? Same question for ViewContext below.